src/neon/utf8check.rs

```diff
@@ -0,0 +1,191 @@
+use crate::neon::intrinsics::*;
+
+/*
+ * legal utf-8 byte sequence
+ * http://www.unicode.org/versions/Unicode6.0.0/ch03.pdf - page 94
+ *
+ *  Code Points        1st       2s       3s       4s
+ * U+0000..U+007F     00..7F
+ * U+0080..U+07FF     C2..DF   80..BF
+ * U+0800..U+0FFF     E0       A0..BF   80..BF
+ * U+1000..U+CFFF     E1..EC   80..BF   80..BF
+ * U+D000..U+D7FF     ED       80..9F   80..BF
+ * U+E000..U+FFFF     EE..EF   80..BF   80..BF
+ * U+10000..U+3FFFF   F0       90..BF   80..BF   80..BF
+ * U+40000..U+FFFFF   F1..F3   80..BF   80..BF   80..BF
+ * U+100000..U+10FFFF F4       80..8F   80..BF   80..BF
+ *
+ */
+
+// all byte values must be no larger than 0xF4
+#[cfg_attr(not(feature = "no-inline"), inline)]
+unsafe fn check_smaller_than_0xf4(current_bytes: int8x16_t, has_error: &mut int8x16_t) {
+    // unsigned, saturates to 0 below max
+    *has_error =
+        vorrq_s8(
+            *has_error,
+            vqsubq_s8(current_bytes, vdupq_n_s8(-12 /* 0xF4 */)));
+}
+
+#[cfg_attr(not(feature = "no-inline"), inline)]
+unsafe fn continuation_lengths(high_nibbles: int8x16_t) -> int8x16_t {
+    let nibbles: int8x16_t = int8x16_t::new(
```

Comment:
I think upstream `consts` this out and used a load instruction to fetch it, perhaps we could/should do the same?

Reply:
sounds good - I was running into some issues with the `vld*` instruction, will pick away at it...